### ISSUE TITLE
Add a migration guide for each major release

### ISSUE THREAD

**Issue by milesfrain** [purescript/documentation#370]
For each major release it would be nice to have a document describing:
- Breaking changes in the compiler (could simply be an excerpt from [release notes](https://github.com/purescript/purescript/releases)).
- New compiler features since the previous major release (just copy all relevant snippets from the minor release notes).
- Breaking changes in core libraries.
- Breaking changes in contrib libraries.

@JordanMartinez has a nice [0.14 guide](https://github.com/JordanMartinez/updatePsLibs/blob/master/ecosystem-changes.md). I think that should live in this docs repo.

Maybe we create a "What's New" (inspired by [F#'s docs](
https://docs.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-50)) folder in this repo. 

The closest thing we currently have are some scattered guides in the [historical](https://github.com/purescript/documentation/tree/master/historical) folder.

**Comment by hdgarrood** [comment]
Agreed 👍 I don’t think we should copy stuff directly from compiler release notes, but rather summarise in a way that’s most helpful to people who don’t care about the details and just want to get their code updated as quickly as possible (as Jordan has done in the existing 0.14 doc).

**Comment by thomashoneyman** [comment]
Halogen has this as well:

https://github.com/purescript-halogen/purescript-halogen/tree/master/docs/changelog
